### PR TITLE
Add area and corresponding test

### PR DESCRIPTION
**Pull request by amanpalariya**
In this pull request area for all the countries have been added in the given format
```js
{
  //...,
  "area": {
    "km2": 1234,
    "mi2": 432,
  }
}
```

Also, a test has been added to `data/tests/test.js` to make sure that area in km<sup>2</sup> and mi<sup>2</sup> match.

### Details about data
This data was obtained from [Wikipedia](https://en.wikipedia.org/wiki/List_of_countries_and_dependencies_by_area) and has been checked for consistency using the test given below.

### Details about test
This test converts the area in km<sup>2</sup> to mi<sup>2</sup> and then checks whether it can be rounded off to match `area.mi2`.

### CODE REVIEW

File: data/tests/test.js [hunk 1]
```diff
@@ -82,6 +82,29 @@ axios.get(flagURL).then((response) => {
...
+      if(diff<=1){
+          numberOfCountriesWithCorrectAreas += 1;
+          break;
+      }
+      roundingPlace += 1;
+      roundedAreaInMi2 = roundTo(expectedAreaInMi2, Math.pow(10, roundingPlace));
+    }
+}
+console.log(`Consistent: ${numberOfCountriesWithCorrectAreas}`);
+console.log(`Total: ${totalCountries}\n`);
```
Comment (sthiepaan) [inline data/tests/test.js:110]: How did you run this to do the testing? To execute all tests that are in the project, you need to run `npm test` in _terminal_. The tests are located in `spec/getCountriesSpec.js`.
Also, I don't understand what is this code doing here, it looks pretty complex tho.

What we need to do in the test is:
1. Describe a test
2. Get all countries
3. Iterate through each `Country {Object}` case
    - Get `area.km2` and `area.mi2`
    - Convert one to another unit
    - Round units to nearest integer
    - Compare whether values are equal

**Review by sthiepaan — CHANGES_REQUESTED** [review]
Thank you for your input @amanpalariya. I added few comments/suggestions to your _PR_ 😉

**Comment by sthiepaan** [comment]
There was a mention in one of our threads that we could also deliver a `getCountriesByAreaRange(from, to, unit)` Method (plus test it for sure). Will you be able to deliver this together with that _Pull Request_? I can help you with that (if necessary)

**Comment by bhatvikrant** [comment]
Thanks alot for the review @sthiepaan! 

@amanpalariya lets resolve the reviews asap and lets get this PR merged!

File: data/tests/test.js (hunk 1 above)
Comment (amanpalariya) [reply data/tests/test.js:110]: ### Details about test
As per the data I found in Wikipedia, some countries' area in mi<sup>2</sup> was rounded to nearest 1000 or 100 (not just the nearest integer). So this test rounds the calculated area until it matches.

Also, this tests disregards an error of ±1 in area.

### Looking for suggestions
I was also wondering why does this `test.js` exist.
Testing of functions is done in `spec/`.

So, I assumed that this file was meant for testing the consistency of the data in `data.json`. Also, the test for flags in `test.js` is not 100% passing.


What should be done? 🤔

Comment (sthiepaan) [reply data/tests/test.js:110]: > As per the data I found in Wikipedia, some countries' area in mi2 was rounded to nearest 1000 or 100 (not just the nearest integer). So this test rounds the calculated area until it matches.

All we need (at least for now), is to have as accurate and precise value as possible. If someone notice that some values are incorrect for some country, we can adjust it.

Important thing is that we should be consistent with rounding and it should work properly for small and big countries (like _Vatican_ 🇻🇦 and _Russia_ 🇷🇺)

What do you think @bhatvikrant, how should we resolve this?

> I was also wondering why does this test.js exist.

Me neither 🤷‍♂️ For now, just ignore this.

---

What would be really great is if you would like to prepare a `getCountriesByAreaRange(from, to)` Method and write tests for it (all tests are located in `spec/getCountriesSpec.js`). After you write a test, you can execute it by running in terminal `npm test` Script.

Comment (amanpalariya) [reply data/tests/test.js:110]: > All we need (at least for now), is to have as accurate and precise value as possible.

Since the data have been copy-pasted from Wikipedia, it would be better to stick to these values instead of changing them ourselves. We can't see anything better than crowd-wisdom of Wikipedia. 🙂

As for the function `getCountriesByAreaRange(from, to)`, I'd open an issue and see if someone would like to take it up.

@bhatvikrant What do you think?

**Comment by bhatvikrant** [comment]
@amanpalariya what's the update on this buddy?

**Comment by amanpalariya** [comment]
> 
> 
> @amanpalariya what's the update on this buddy?

I have added the data as is from Wikipedia in the decided format.

Some values in mi<sup>2</sup> did not match to that in km<sup>2</sup> because they were rounded off (in Wikipedia itself).

Damian says that we must make the values accurate. But I think we should stick with what's written on Wikipedia.

As for the `getCountriesByAreaRange` method, I think I will open an issue for that.

**TL;DR**
This PR is ready for merge if you are okay with data on Wikipedia.

**Comment by bhatvikrant** [comment]
> > @amanpalariya what's the update on this buddy?
> 
> @bhatvikrant
> 
> I have added the data as is from Wikipedia in the decided format.
> 
> Some values in mi2 did not match to that in km2 because they were rounded off (in Wikipedia itself).
> 
> Damian says that we must make the values accurate. But I think we should stick with what's written on Wikipedia.
> 
> As for the `getCountriesByAreaRange` method, I think I will open an issue for that.
> 
> **TL;DR**
> This PR is ready for merge if you are okay with data on Wikipedia.

@amanpalariya please resolve the conflicts while I am looking into the unit issue

**Comment by amanpalariya** [comment]
@bhatvikrant The merge conflicts have been resolved.

Sorry for the delay.

**Review by sthiepaan — CHANGES_REQUESTED** [review]
Seems like we still struggle with some merge conflicts. Could you resolve conflicts on your _Pull Request_ @amanpalariya? I would like to merge this _PR_ as next 😉

**Comment by sthiepaan** [comment]
@amanpalariya I cannot merge this _Pull Request_ because it has conflicts. You need to resolve conflicting files. In this _PR_ case that is `spec/getCountriesSpec.js`.

**Review by sthiepaan — APPROVED** [review]
LGTM 🎉
